### PR TITLE
Add Slider View  💥💥

### PR DESCRIPTION
**Pull request by HebaMad**

### CODE REVIEW

File: Dokan/Dokan/Scenes/ProductDetails/SliderView/SliderViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//  SliderViewModel.swift
+//  Dokan
+//
+//  Created by heba isaa on 10/08/2022.
+//
+
+import Foundation
+import UIKit
+
+struct sliderViewModel {
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/SliderView/SliderViewModel.swift]: I would prefer adding this view model as a nested type in `SliderCollectionViewCell`

Example: 
```swift
extension SliderCollectionViewCell { 
    struct ViewModel { 
        let productImage: UIImage
        let noOfPictures: Int
        let noOfCurrentPicture: Int
    }
}
```

File: Dokan/Dokan/Scenes/ProductDetails/SliderView/SliderView.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    func fixInView(_ container: UIView) {
+        translatesAutoresizingMaskIntoConstraints = false
+        frame = container.frame
+        container.addSubview(self)
+        NSLayoutConstraint(item: self, attribute: .leading, relatedBy: .equal, toItem: container, attribute: .leading, multiplier: 1.0, constant: 0).isActive = true
+        NSLayoutConstraint(item: self, attribute: .trailing, relatedBy: .equal, toItem: container, attribute: .trailing, multiplier: 1.0, constant: 0).isActive = true
+        NSLayoutConstraint(item: self, attribute: .top, relatedBy: .equal, toItem: container, attribute: .top, multiplier: 1.0, constant: 0).isActive = true
+        NSLayoutConstraint(item: self, attribute: .bottom, relatedBy: .equal, toItem: container, attribute: .bottom, multiplier: 1.0, constant: 0).isActive = true
+    }
+}
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/SliderView/SliderView.swift]: I think we already have a similar method. If not, please add it to `DokanUI` helpers.

File: Dokan/Dokan/Scenes/ProductDetails/SliderView/SliderView.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        productSliderCollectionView.dataSource = self
+        collectionViewLayout()
+    }
+
+    func collectionViewLayout() {
+        let layout = UICollectionViewFlowLayout()
+        layout.sectionInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+        layout.itemSize = CGSize(width: (productSliderCollectionView.bounds.width) - 5, height: productSliderCollectionView.bounds.height)
+        layout.minimumInteritemSpacing = 15
+        layout.minimumLineSpacing = 15
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/SliderView/SliderView.swift]:
```suggestion
        layout.minimumInteritemSpacing = 16
        layout.minimumLineSpacing = 16
```

File: Dokan/Dokan/Scenes/ProductDetails/SliderView/SliderView.swift [hunk 4]
```diff
@@ -0,0 +1,87 @@
+//
+//  SliderView.swift
+//  Dokan
+//
+//  Created by heba isaa on 10/08/2022.
+//
+
+import UIKit
+
+class SliderView: UIView {
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/SliderView/SliderView.swift:10]: Could you please integrate SliderView to `ProductDetailsViewController`?

**Review by ahmdmhasn — APPROVED** [review]
Great progress 💪

**Review by oradwanomar — APPROVED** [review]

**Pull request merged**